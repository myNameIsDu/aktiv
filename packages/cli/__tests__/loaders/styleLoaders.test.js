const styleLoaders = require('../../ak-webpack-config/lib/loaders/styleLoaders');

describe('styleLoaders快照', () => {
    it('extractCSS为false', () => {
        // @ts-ignore 仅使用到extractCss
        const loaders = styleLoaders({ extractCSS: false });

        expect(loaders).toMatchSnapshot();
    });

    it('extractCSS为true', () => {
        // @ts-ignore 仅使用到extractCss
        const loaders = styleLoaders({ extractCSS: true });

        expect(loaders).toMatchSnapshot();
    });
});
