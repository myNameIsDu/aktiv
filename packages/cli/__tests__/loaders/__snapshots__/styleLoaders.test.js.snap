// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`styleLoaders快照 extractCSS为false 1`] = `
Array [
  Object {
    "test": /\\\\\\.css\\$/,
    "use": Array [
      "style-loader",
      "css-loader",
      Object {
        "loader": "postcss-loader",
        "options": Object {
          "postcssOptions": Object {
            "plugins": Array [
              [Function],
            ],
          },
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.less\\$/,
    "use": Array [
      "style-loader",
      "css-loader",
      Object {
        "loader": "postcss-loader",
        "options": Object {
          "postcssOptions": Object {
            "plugins": Array [
              [Function],
            ],
          },
        },
      },
      Object {
        "loader": "less-loader",
        "options": Object {
          "javascriptEnabled": true,
        },
      },
    ],
  },
]
`;

exports[`styleLoaders快照 extractCSS为true 1`] = `
Array [
  Object {
    "test": /\\\\\\.css\\$/,
    "use": Array [
      Object {
        "loader": "/Users/du/Documents/my/my-cli/du-formula/node_modules/mini-css-extract-plugin/dist/loader.js",
      },
      "css-loader",
      Object {
        "loader": "postcss-loader",
        "options": Object {
          "postcssOptions": Object {
            "plugins": Array [
              [Function],
            ],
          },
        },
      },
    ],
  },
  Object {
    "test": /\\\\\\.less\\$/,
    "use": Array [
      Object {
        "loader": "/Users/du/Documents/my/my-cli/du-formula/node_modules/mini-css-extract-plugin/dist/loader.js",
      },
      "css-loader",
      Object {
        "loader": "postcss-loader",
        "options": Object {
          "postcssOptions": Object {
            "plugins": Array [
              [Function],
            ],
          },
        },
      },
      Object {
        "loader": "less-loader",
        "options": Object {
          "javascriptEnabled": true,
        },
      },
    ],
  },
]
`;
